hack/jenkins/e2e-runner.sh

```diff
@@ -21,96 +21,124 @@ set -o nounset
-        # Otherwise, we want a completely empty directory.
-        if [[ "${JENKINS_FORCE_GET_TARS:-}" =~ ^[yY]$ ]]; then
-            rm -rf kubernetes*
-        # .config and its children are created by the gcloud call that we use to
-        # get the GCE service account.
-        # console-log.txt is created by Jenkins, but is usually not flushed out
-        # this early in the script.
-        elif [[ $(find . -not -path "./.config*" -not -name "console-log.txt" \
-                  | wc -l) != 1 ]]; then
-            echo "${PWD} not empty, bailing!"
-            find .
-            exit 1
-        fi
-
-        # This is for test, staging, and prod jobs on GKE, where we want to
-        # test what's running in GKE by default rather than some CI build.
-        if [[ ${JENKINS_USE_SERVER_VERSION:-}  =~ ^[yY]$ ]]; then
-            # for GKE we can use server default version.
-            bucket="release"
-            msg=$(gcloud ${CMD_GROUP} container get-server-config --project=${PROJECT} --zone=${ZONE} | grep defaultClusterVersion)
-            # msg will look like "defaultClusterVersion: 1.0.1". Strip
-            # everything up to, including ": "
-            build_version="v${msg##*: }"
-            echo "Using server version $bucket/$build_version"
-        else  # use JENKINS_PUBLISHED_VERSION, for CI
-            # Use a published version like "ci/latest" (default),
-            # "release/latest", "release/latest-1", or "release/stable"
-            JENKINS_PUBLISHED_VERSION=${JENKINS_PUBLISHED_VERSION:-'ci/latest'}
-            IFS='/' read -a varr <<< "${JENKINS_PUBLISHED_VERSION}"
-            bucket="${varr[0]}"
-            build_version=$(gsutil cat gs://kubernetes-release/${JENKINS_PUBLISHED_VERSION}.txt)
-            echo "Using published version $bucket/$build_version (from ${JENKINS_PUBLISHED_VERSION})"
-            # Set CLUSTER_API_VERSION for GKE CI
-            export CLUSTER_API_VERSION=$(echo ${build_version} | cut -c 2-)
-        fi
-        # At this point, we want to have the following vars set:
-        # - bucket
-        # - build_version
-        gsutil -mq cp \
-            "gs://kubernetes-release/${bucket}/${build_version}/kubernetes.tar.gz" \
-            "gs://kubernetes-release/${bucket}/${build_version}/kubernetes-test.tar.gz" \
-            .
+if [[ "${JENKINS_USE_EXISTING_BINARIES:-}" =~ ^[yY]$ ]]; then
+    echo "Tolerating dirty workspace (because JENKINS_USE_EXISTING_BINARIES)"
+elif [[ "${JENKINS_TOLERATE_DIRTY_WORKSPACE:-}" =~ ^[yY]$ ]]; then
+    echo "Tolerating dirty workspace (because JENKINS_TOLERATE_DIRTY_WORKSPACE)"
+else
+    # .config and its children are created by the gcloud call that we use to
+    # get the GCE service account.
+    # console-log.txt is created by Jenkins, but is usually not flushed out
+    # this early in the script.
+    #
+    # TODO(ihmccreery) we haven't gotten any service accounts yet, so we should
```

Comment:
Oh dear; that we are calling gcloud from shell inside YAML makes me cringe.  The stuff in YAML should *only* export vars.

Fixed.